src/core/instance/instance.cpp

```diff
@@ -342,7 +342,7 @@ uint8_t Instance::GetIdx(Instance *aInstance)
 {
     return static_cast<uint8_t>(
         (reinterpret_cast<uint8_t *>(aInstance) - reinterpret_cast<uint8_t *>(gMultiInstanceRaw)) /
-        INSTANCE_SIZE_ALIGNED);
+        (INSTANCE_SIZE_ALIGNED * sizeof(uint64_t)));
```

Comment:
Can we consider instead doing 
`(reinterpret_cast<uint64_t *>(aInstance) - gMultiInstanceRaw) / INSTANCE_SIZE_ALIGNED`?

This would make it consistent with how `Get(aIdx)` seems to work (which does the pointer arithmetic using `gMultiInstanceRaw` directly, which is treated as a `(uint64_t *)`.

Reply:
Thank you, yes, this will work.